src/ArmTemplates/Common/Extensions/ResourceNamingHelper.cs

```diff
@@ -0,0 +1,26 @@
+﻿// --------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+// --------------------------------------------------------------------------
+
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Extensions
+{
+    public static class ResourceNamingHelper
```

Comment:
Duplicate with [ParameterNamingHelper](https://github.com/Azure/azure-api-management-devops-resource-kit/blob/main/src/ArmTemplates/Common/Extensions/ParameterNamingHelper.cs). Or do I miss anything?

Reply:
I didn't want to change the already existing ParameterNamingHelper since related to the parameter naming, and for the resourceNaming helper it has different rules for example it could have in the name '-' char. Also, in the future each of this classes could be extended with resource/parameter naming functionality.

Reply:
Combined the Parameter and Resource naming helpers in one NamingHelper.